src/NServiceBus.RabbitMQ/Sending/BasicPropertiesExtensions.cs

```diff
@@ -13,15 +13,18 @@ static class BasicPropertiesExtensions
     {
         public static void Fill(this IBasicProperties properties, OutgoingMessage message, List<DeliveryConstraint> deliveryConstraints)
         {
-            properties.MessageId = message.MessageId;
+            if (message.MessageId != null)
+            {
+                properties.MessageId = message.MessageId;
```

Comment:
I was initially confused as to why we need to do this, but then I decompiled the source and the side effect revealed itself: 

```C#
// RabbitMQ.Client.Framing.BasicProperties
public override string MessageId
{
    get
    {
        return this.m_messageId;
    }
    set
    {
        this.m_messageId_present = true;
        this.m_messageId = value;
    }
}
```

Note that I can't link to the source because the class is autogenerated 😖